### PR TITLE
Features/get course

### PR DESCRIPTION
**Pull request by chamal817**

### CODE REVIEW

File: src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Functions/CourseFunctions.cs [hunk 1]
```diff
@@ -50,30 +50,48 @@ public async Task<HttpResponseData> CreateCourse([HttpTrigger(AuthorizationLevel
...
 
-            if (string.IsNullOrWhiteSpace(pageSize) || !int.TryParse(pageSize, out var size) || size <= 0)
+                if (courses == null || !courses.Items.Any())
+                {
+                    _logger.LogWarning("No data.");
+                    response = req.CreateResponse(HttpStatusCode.NotFound);
+                    await response.WriteAsJsonAsync("No data");
+                    return response;
+                }
+                else
```
Comment (gayashan4lk) [inline src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Functions/CourseFunctions.cs:84]: don't need else here.

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx [hunk 2]
```diff
@@ -64,3 +64,26 @@ export function useComponentState() {
 
   return { formik, deleteCourse };
 }
+
+export function useCoursesState() {
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx:68]: don't create a new hook here. just create a function which responsible for fetch courses. Also define api functions in the api.ts file.

File: src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Functions/CourseFunctions.cs (hunk 1 above)
Comment (gayashan4lk) [inline src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Functions/CourseFunctions.cs:84]: don't need the else block here.

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx (hunk 2 above)
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx:68]: don't create a new hook here. just create a function which responsible for fetch courses. Also define api functions in the api.ts file.

**Review by gayashan4lk — CHANGES_REQUESTED** [review]
check comments.

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx [hunk 3]
```diff
@@ -56,7 +57,14 @@ export default function CoursesContainer() {
       },
     },
   ];
+  
+  useEffect(() => {
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx:61]: move this useEffect in to state.tsx

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx [hunk 4]
```diff
@@ -62,5 +63,19 @@ export function useComponentState() {
     onSubmit: createNewCourse,
   });
 
-  return { formik, deleteCourse };
+  const fetchCourses = async (pageNumber: number, pageSize: number) => {
+    let courseRes: CourseResponse[] = [await getCourse(pageNumber, pageSize)];
+    if (Array.isArray(courseRes)) {
+      courseRes = courseRes.flat();
+    }
+    const formattedData = courseRes.map((item) => {
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx:71]: add a key for each element of the mapped array.

File: src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export enum Pagination{
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts:1]: better to rename as "DefaultPagination"

**Review by gayashan4lk — CHANGES_REQUESTED** [review]

File: src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Functions/CourseFunctions.cs (hunk 1 above)
Comment (chamal817) [reply src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Functions/CourseFunctions.cs:84]: fixed

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx (hunk 2 above)
Comment (chamal817) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx]: fixed

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx (hunk 3 above)
Comment (chamal817) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx]: fixed

File: src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts (hunk 5 above)
Comment (chamal817) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts]: fixed

File: src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export enum DefaultPagination{
+  pageNumber = 1,
+  pageSize = 10,
+
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts:4]: remove empty line

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx [hunk 7]
```diff
@@ -62,5 +64,24 @@ export function useComponentState() {
...
+      const formattedEndDate = new Date(item.EndDate).toLocaleDateString();
+      return { ...item, StartDate: formattedStartDate, EndDate: formattedEndDate,key: item.Id};
+    });
+    setDataSource(formattedData);
+  };
+
+
+  return { formik, deleteCourse, dataSource, fetchCourses };
 }
+export const useFetchCourseEffect = (fetchCourses) => {
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx:83]: move this into the `useComponentState`

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx [hunk 8]
```diff
@@ -56,7 +57,10 @@ export default function CoursesContainer() {
       },
     },
   ];
-
+  useFetchCourseEffect(fetchCourses)
```
Comment (gayashan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx:60]: don't need to use this here. Fetch courses in `state.ts` and just get the result to this file.

**Review by gayashan4lk — CHANGES_REQUESTED** [review]

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx (hunk 8 above)
Comment (chamal817) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/CoursesContainer.tsx]: fixed

File: src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx (hunk 7 above)
Comment (chamal817) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/components/courses-container/state.tsx]: fixed

File: src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts (hunk 6 above)
Comment (chamal817) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/interfaces/enums.ts]: fixed

**Review by gayashan4lk — APPROVED** [review]

**Pull request merged**